modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/ColumnDefinition.java

```diff
@@ -71,13 +72,13 @@ public boolean nullable() {
      * @return Column's precision.
      */
     public Integer precision() {
-        return type.getPrecision() != RelDataType.PRECISION_NOT_SPECIFIED ? type.getPrecision() : null;
+        return TypeUtils.hasPrecesion(type) ? type.getPrecision() : null;
```

Comment:
What if we return -1 here?

Reply:
The behavior will be the same (empty precision and scale will be converted to -1 finally), but I don't think it's a good idea:
1. It requires a little bit more memory for intermediate HashMaps
2. It's a magic number (normally we shouldn't know about meaning of -1 for query entity, perhaps it will have another meaning in Ignite 3)